### PR TITLE
Feature/windows support

### PR DESCRIPTION
**Pull request by mjheilmann**
This is a modification from the capnp fork with QNX changes, that this is PR'd into.

This contains (or should not contain) any functional changes, but only symbol visibility, and a few hacks for windows support.  this has been tested successfully with LumPDK, and the FW ReSIM library.

This has a few choice `#undef` lines for LumPDK windows compat, as there is currently not a simple way to isolate the windows import and offending `#define` lines from the windows headers;  while this is not ideal it should not be harmful for our intentions.

### CODE REVIEW

File: c++/src/capnp/schema.h [hunk 1]
```diff
@@ -56,20 +59,20 @@ using SchemaType = typename SchemaType_<T>::Type;
 // SchemaType<T> is the type of T's schema, e.g. StructSchema if T is a struct.
 
 namespace _ {  // private
-extern const RawSchema NULL_SCHEMA;
-extern const RawSchema NULL_STRUCT_SCHEMA;
-extern const RawSchema NULL_ENUM_SCHEMA;
-extern const RawSchema NULL_INTERFACE_SCHEMA;
-extern const RawSchema NULL_CONST_SCHEMA;
+CAPNP_API constexpr const RawSchema const* NULL_SCHEMA();
```
Comment (radicaljims) [inline c++/src/capnp/schema.h]: `gcc` complains when compiling for QNX:

```
In file included from /home/james/work/LumPDK_qnx_dependencies/capnproto/c++/src/capnp/capability.h:35:0,
                 from /home/james/work/LumPDK_qnx_dependencies/capnproto/c++/src/capnp/arena.h:43,
                 from /home/james/work/LumPDK_qnx_dependencies/capnproto/c++/src/capnp/layout.c++:25:
/home/james/work/LumPDK_qnx_dependencies/capnproto/c++/src/capnp/schema.h:62:37: error: duplicate 'const'
 CAPNP_API constexpr const RawSchema const* NULL_SCHEMA();
```
After a quick google I tried switching the line to

```
CAPNP_API constexpr const RawSchema *const NULL_SCHEMA();
```

and updated the corresponding definition but that lead to `gcc` thinking this was an inline function without a definition.

Not sure if that's even a correct path to go down. Any suggestions?

Comment (mjheilmann) [reply c++/src/capnp/schema.h]: we talked on slack, but for a complete record, I made that change and a few others which seemed to be working, or at least move the problems further in the compilation process.

Comment (mjheilmann) [reply c++/src/capnp/schema.h]: `constexpr` can (does?) imply `inline` at least through gcc, and I assume QNX compilers as well.  These functions may not be `constexpr` and not have their definition located with the declaration in the header.

I have pushed what should be fixes.   x86 linux now compiles without warnings using `mkdir build && cd build && cmake .. && cmake --build .` from inside the `c++` directory.  It continues to compile and function on Windows.

Comment (radicaljims) [reply c++/src/capnp/schema.h]: With these changes it compiles on `QNX` again as well. :+1:

**Review by radicaljims — APPROVED** [review]

**Pull request merged**